gc3apps/bf.uzh.ch/gcsvBulk.py

```diff
@@ -0,0 +1,216 @@
+    is executed on the remote end.
+    """
+    application_name = 'gcsvBulk'
+
+    def __init__(self, input_file, input_function, input_csvfile, limit, skip, **extra_args):
+
+        self.output_dir = extra_args['output_dir']
+        self.input_function = input_function
+        #
+        inputs = dict()
+
+        gcsvBulk_wrapper_m = resource_filename(Requirement.parse("gc3pie"),
+                                              "gc3libs/etc/gcsvBulk_wrapper.m")
+        #
+        inputs[gcsvBulk_wrapper_m] = os.path.basename(gcsvBulk_wrapper_m)
+        # inputs['foo'] = 'foo'
+        #
+        inputs[input_file] = "%s" % os.path.basename(input_file)
+        inputs[input_csvfile] = "%s" % os.path.basename(input_csvfile)
+
+        for file in os.listdir(os.getcwd()):
+            if file.endswith(".m"):
+                inputs[file] = "%s" % os.path.basename(file)
+        #
+        arguments = "matlab -nodesktop -nodisplay -nojvm -nosplash -r \"%s %s %s %d %d;exit();\"" % (inputs[gcsvBulk_wrapper_m][:-2], input_function, inputs[input_csvfile], limit, skip)
+        # print 'hello'
+        Application.__init__(
+            self,
+            arguments = arguments,
+            inputs = inputs,
+            outputs = gc3libs.ANY_OUTPUT,
+            stdout = 'gcsvBulk.log',
+            join=True,
+            executables = "./%s" % os.path.basename(input_function),
+            **extra_args)
+
+class GcsvBulkScript(SessionBasedScript):
+    """
+    The script takes as input a mfunction, the number of parallel jobs to run and the
+    dataset to be processed
+
+    The ``gcsvBulk`` command keeps a record of jobs (submitted, executed
+    and pending) in a session file (set name with the ``-s`` option); at
+    each invocation of the command, the status of all recorded jobs is
+    updated, output from finished jobs is collected, and a summary table
+    of all known jobs is printed.
+
+    Options can specify a maximum number of jobs that should be in
+    'SUBMITTED' or 'RUNNING' state; ``gcsvBulk`` will delay submission of
+    newly-created jobs so that this limit is never exceeded.
+    """
+
+    def __init__(self):
+        SessionBasedScript.__init__(
+            self,
+            version = __version__, # module version == script version
+            application = GcsvBulkApplication,
+            # only display stats for the top-level policy objects
+            # (which correspond to the processed files) omit counting
+            # actual applications because their number varies over
+            # time as checkpointing and re-submission takes place.
+            stats_only_for = GcsvBulkApplication,
+            )
+
+    def setup_args(self):
+        self.add_param('mfunction', help=(
+            "Name of the matlab function to run."
+            "A corresponding `.m` file must exist in the current directory."))
+        self.add_param('csvfile', help=(
+            "Name of the csv filte to be processed."))
```

Comment:
Typo: filte -> file

Also see above for ensuring that a path names an existing file.